### ISSUE TITLE
Question: How is MSBuild actually called?

### ISSUE THREAD

**Issue by borgdylan** [dotnet/cli#8301]
I am still aiming to swap the MsBuild that ships with the CLI with the system one (which works for me when used by itself). Using msbuild by itself means no helper commands like sln, test etc. Using the CLI as it gets installed means no mono support. Is there an environment variable I can change or a file I can change to point to the system wide msbuilf that I have installed?

**Comment by borgdylan** [comment]
From my research, seems like a dotnet exec is being done on MSBuild core directly. I plan to write a small .NET core app that bootstraps to the system MSBuild install.

**Comment by kasper3** [comment]
Does dotnet-msbuild invokes

* `MSBuild/<Version>/Bin/amd64/MSBuild.exe` when dotnet-cli is 64 bits
* `MSBuild/<Version>/Bin/MSBuild.exe` when dotnet-cli is 32 bits

OR

* 64-bit `MSBuild.dll` from sharedfx directory when dotnet-cli is 64 bits
* 32-bit `MSBuild.dll` from sharedfx directory when dotnet-cli is 32 bits

OR

Always uses 32-bit `MSBuild.dll` or `MSBuild.exe`?

**Comment by livarcocc** [comment]
The CLI bundles with it the core version of msbuild and it always invokes that version when we need to fire msbuild up.

There is not environment variable or other mechanism to change that, as unfortunately, there are many parts of the system that rely on where msbuild is located in order for the CLI to work properly.

We don't have any short term places to enable this scenario. When we poked at it a while ago we realized that there are more things tangled around it than we would like to and than we anticipated, unfortunately.
